src/app/downloads/downloads.component.ts

```diff
@@ -19,211 +10,22 @@ import { isPodcastChanged, isBeginDateChanged, isEndDateChanged, isIntervalChang
     <section *ngIf="loaded$ | async">
       <metrics-menu-bar></metrics-menu-bar>
       <metrics-downloads-chart></metrics-downloads-chart>
-      <metrics-downloads-table [totalPages]="totalPages"></metrics-downloads-table>
+      <metrics-downloads-table></metrics-downloads-table>
       <p class="error" *ngFor="let error of errors$ | async">{{error}}</p>
     </section>
   `,
   styleUrls: ['downloads.component.css']
 })
-export class DownloadsComponent implements OnInit, OnDestroy {
-  podcastSub: Subscription;
-  podcasts: PodcastModel[];
-  podcast: PodcastModel;
-  episodeSub: Subscription;
-  pageEpisodes: EpisodeModel[];
-  totalPages: number;
-  routerSub: Subscription;
-  routerState: RouterModel;
-  updatePodcast: boolean;
-  updateEpisodes: boolean;
+export class DownloadsComponent implements OnInit {
   loading$: Observable<boolean>;
   loaded$: Observable<boolean>;
   errors$: Observable<string[]>;
 
-  constructor(private castle: CastleService,
-              public store: Store<any>,
-              private router: Router) {
+  constructor(public store: Store<any>) {}
+
+  ngOnInit() {
     this.loading$ = this.store.pipe(select(selectLoading));
     this.loaded$ = this.store.pipe(select(selectLoaded));
     this.errors$ = this.store.pipe(select(selectErrors));
   }
-
-  ngOnInit() {
-    this.subPodcastsAndRoute();
-  }
-
-  subPodcastsAndRoute() {
-    this.podcastSub = this.store.pipe(select(selectPodcasts)).subscribe((podcasts: PodcastModel[]) => {
-      if (podcasts && podcasts.length) {
-        this.podcasts = podcasts;
-
-        if (!this.routerSub) {
-          this.routerSub = this.store.pipe(select(selectRouter)).subscribe((newRouterState: RouterModel) => {
```

Comment:
This component used to watch for changes and load data, but now that happens in the routing service. So this component is only responsible for displaying the on loading and displaying the app components when loaded. Lean components are 🔥